sr_gui_change_controllers/src/sr_gui_change_controllers/change_controllers.py

```diff
@@ -153,12 +182,34 @@ def confirm_current_control(self):
                         current_robot_control[robot_name] = 0
                         break
                     elif "effort_controller" in controller.name:
-                        current_robot_control[robot_name] = 2
+                        if "h_" in robot_name:
+                            if all(hand in self._controller_groups for hand in ["rh_", "lh_"]):
+                                hand_id = 'sr_bimanual_hands_robot'
+                            else:
+                                hand_id = 'sr_hand_robot'
+                            srv_path = '/{}/{}/change_control_type'.format(hand_id, robot_name[:2])
+                            query_control_type = rospy.ServiceProxy(srv_path, ChangeControlType)
+                            try:
+                                change_type_msg = ChangeControlType()
+                                change_type_msg.control_type = ControlType.QUERY
+                                resp1 = query_control_type(change_type_msg)
+
+                                if resp1.result.control_type == 0:
+                                    current_robot_control[robot_name] = 3
+                                elif resp1.result.control_type == 1:
+                                    current_robot_control[robot_name] = 2
+                            except rospy.ServiceException as err:
+                                error = "Couldn't get control type from {} service".format(srv_path)
+                                QMessageBox.warning(self._widget, "No Control Type Found", error)
+                                rospy.logerr(error + ". Error: " + err)
+                                return
+                        else:
+                            current_robot_control[robot_name] = 2
```

Comment:
and here

Reply:
fixed in [542d322](https://github.com/shadow-robot/sr_visualization_common/pull/35/commits/542d322c257524753a6d7f478c22fe42ffc70d69)